app/main.py

```diff
@@ -109,11 +109,9 @@ async def knowledge_panel(
             # we will get it on value retrieval
             # but we don't want to sacrifice whole result for a single failure
             # as most panels depends on external resources that may not be available
-            try:
-                if soon_value.value:
-                    panels.update(soon_value.value)
-            except Exception as e:
-                logging.exception(msg=e)
+
+            if soon_value.value:
+                panels.update(soon_value.value)
```

Comment:
Did you investigated why this try/except did not get exceptions ?
I'm a bit reluctant to loose this pattern for it was more robust and clean to have it here.

Reply:
It seems "task_group.soonify" doesn't catch the exceptions.

Reply:
according to the docs, it should trigger it as we ask the value, but maybe we are doing something wrong (maybe we should stay in the "with" ?

I would try to simply indent the code to keep it under "with". That would be the only change you have to do to the whole code.

Reply:
@sumit-158 I will try to look into this next week. It intrigues me !

Reply:
You are right https://github.com/tiangolo/asyncer/blob/main/asyncer/_main.py#L153

That's a bit sad !

So maybe we should add a decorator around our functions to catch any exception.

```
@no_fail
async def data_quality_kp(self, ...
```

Do you want to try to write such an async decorator ?

Reply:
@alexgarel made the function, please do check it :-)